src/index.js

```diff
@@ -1,130 +1,236 @@
+		return runSpecs( {
+			specs: specs,
+			coverageExclude: coverageExclude,
+			coverage: _opt.coverage
+		} );
+	}
+
+	function testBehaviorAndExit( opt ) {
+		return testBehavior( opt )
+			.on( "end", process.exit.bind( process, 0 ) )
+			.on( "error", process.exit.bind( process, 100 ) );
+	}
+
+	function testIntegration( opt ) {
+		var _opt = opt || options;
+		var specs = permuPath( _opt.integrationSpecPaths, _opt.specs );
+		var coverageExclude = permuPath( _opt.allSpecPaths, [ "**/*" ]);
+		return runSpecs( {
+			specs: specs,
+			coverageExclude: coverageExclude,
+			coverage: _opt.coverage
+		} );
+	}
+
+	function testIntegrationAndExit( opt ) {
+		return testIntegration( opt )
+			.on( "end", process.exit.bind( process, 0 ) )
+			.on( "error", process.exit.bind( process, 100 ) );
+	}
+
+	function cover( opt ) {
+		var _opt = opt || _.clone(options);
+		_opt.coverage = true;
+		return testAll( _opt );
+	}
+
+	function showCoverage( opt ) {
+		var _opt = opt || _.clone(options);
+		_opt.coverage = true;
+		return testAll( opt ).on( "end", function() {
+			open( _opt.coverageHtml );
+			process.exit();
+		} );
+	}
+
+	function setupProcess( processName, opts ) {
+		processesDefined = true;
+		return processhost.startProcess( processName, {
+			command: opts.command || opts.cmd || "node" + cmdPostfix,
+			args: opts.arguments || opts.args || [],
+			stdio: opts.stdio || "inherits",
+			restart: opts.restart
+		} );
+	}
+
+	function setupProcesses( opts ) {
+		processesDefined = true;
+		return processhost.setup( opts );
+	}
+
+	function groklate( ngrok, opts ) {
+		if(!opts && !options.ngrok) {
+			throw new Error("To use ngrok you must supply an options argument containing port, subdomain and token");
+		}
+		var _opt = opts || options.ngrok;
+		return when.promise( function( resolve, reject ) {
+			ngrok.connect( {
+				port: _opt.port,
+				subdomain: _opt.subdomain,
+				authtoken: _opt.token
+			}, function( err, url ) {
+					if ( err ) {
+						reject( err );
+					} else {
+						resolve( url );
+					}
+				} );
+		} );
+	}
+
+	function watch( opt, tasks ) {
+		var _tasks;
+		var _opt;
+		if ( arguments.length === 1 ) {
+			_opt = options;
+			_tasks = opt;
+		} else {
+			_opt = opt;
+			_tasks = tasks;
+		}
+		if ( processesDefined ) {
+			_tasks.concat( "restartProcesses" );
+		}
+		return gulp.watch( _opt.watchPaths, _tasks );
+	}
+
+	function format( opt ) {
+		var _opt = opt || options;
+		return gulp.src( _opt.sources.concat( _opt.exclude ) )
+			.pipe( jscs( {
+				configPath: _opt.jscsCfgPath,
+				fix: true
+			} ) )
+			.on( "error", function( error ) {
+				gutil.log( gutil.colors.red( error.message ) );
+				this.end();
+			} )
+			.pipe( gulpChanged( ".", { hasChanged: gulpChanged.compareSha1Digest } ) )
+			.pipe( gulp.dest( "." ) );
+	}
+
+	function lint( opt ) {
+		var _opt = opt || options;
+		return gulp.src( permuPath( _opt.sourcePaths, _opt.sources ) )
```

Comment:
It looks like this is missing the specs and gulpfile when linting. Maybe use: `_opt.sources.concat( _opt.exclude )` as you do in `format`